### PR TITLE
[MRESOLVER-151] Enforce a checksum policy to be provided explicitly

### PR DESCRIPTION
**Pull request by michael-o**

### CODE REVIEW

File: maven-resolver-impl/src/test/java/org/eclipse/aether/internal/impl/DefaultChecksumPolicyProviderTest.java [hunk 1]
```diff
@@ -124,7 +124,7 @@ public void testGetEffectiveChecksumPolicy_DifferentPolicies()
         }
     }
 
-    @Test
+    @Test( expected = IllegalArgumentException.class )
```
Comment (rfscholte) [inline maven-resolver-impl/src/test/java/org/eclipse/aether/internal/impl/DefaultChecksumPolicyProviderTest.java:127]: With the change it'll exit with the first testcase and the first assert, making the rest useless.
Better replace every `assertEquals` with something like

    IllegalArgumentException ex = assertThrows( IllegalArgumentException.class, () -> provider.getEffectiveChecksumPolicy(.....) );
    assertThat( ex.getMessage(), is(...))

**Review by rfscholte — CHANGES_REQUESTED** [review]

File: maven-resolver-impl/src/test/java/org/eclipse/aether/internal/impl/DefaultChecksumPolicyProviderTest.java (hunk 1 above)
Comment (michael-o) [reply maven-resolver-impl/src/test/java/org/eclipse/aether/internal/impl/DefaultChecksumPolicyProviderTest.java:127]: Yepp, that's true. Will change!

File: maven-resolver-impl/src/test/java/org/eclipse/aether/internal/impl/DefaultChecksumPolicyProviderTest.java [hunk 2]
```diff
@@ -133,10 +135,12 @@ public void testGetEffectiveChecksumPolicy_UnknownPolicies()
                 { RepositoryPolicy.CHECKSUM_POLICY_IGNORE, RepositoryPolicy.CHECKSUM_POLICY_IGNORE } };
         for ( String[] testCase : testCases )
         {
-            assertEquals( "unknown vs " + testCase[1], testCase[0],
-                          provider.getEffectiveChecksumPolicy( session, CHECKSUM_POLICY_UNKNOWN, testCase[1] ) );
-            assertEquals( "unknown vs " + testCase[1], testCase[0],
-                          provider.getEffectiveChecksumPolicy( session, testCase[1], CHECKSUM_POLICY_UNKNOWN ) );
+            IllegalArgumentException e = assertThrows( IllegalArgumentException.class,
+                    () -> provider.getEffectiveChecksumPolicy( session, CHECKSUM_POLICY_UNKNOWN, testCase[1] ) );
+            assertThat( e.getMessage(), is("Unsupported policy: unknown") );
```
Comment (rfscholte) [inline maven-resolver-impl/src/test/java/org/eclipse/aether/internal/impl/DefaultChecksumPolicyProviderTest.java:140]: I didn't notice the static import of `assertThrows`, which is caused by an existing wildcard import. Please make imports explicit.

Comment (michael-o) [reply maven-resolver-impl/src/test/java/org/eclipse/aether/internal/impl/DefaultChecksumPolicyProviderTest.java:140]: OK, all asserts where implicitly imported in this class. I will turn them into explicit ones.

**Review by rfscholte — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #860

**Pull request closed without merging**